### PR TITLE
Add a migration notice on certain pages

### PR DESCRIPTION
**Pull request by kienstra**
#### Changes
* Adds the migration and 'Not Now' notices
* On clicking 'Not Now', the migration notice never appears again for that user:
* Appears on Block Lab > Content Blocks, Block Lab > Settings, `/wp-admin/plugins.php`, and the Dashboard. 

#### Testing instructions
1. Go to Block Lab > Content Blocks, Block Lab > Settings, `/wp-admin/plugins.php`, and the Dashboard
2. Expected: the notice appears as shown above
3. Click 'Not Now'
4. Expected: the notice is hidden, and the 'Not Now' notice displays
5. Reload the page
6. Expected: The notice doesn't appear again
7. To make the notice appear again:
```sh
wp user list # Find your user ID from that
wp shell
update_user_meta( $user_id, 'block_lab_show_migration_notice', 1 ); // $user_id is the ID from the wp user list command.
```

### CODE REVIEW

File: js/admin.migration-notice.js
Comment (kienstra) [inline js/admin.migration-notice.js:3]: Haha, it's strange writing vanilla JS again

File: php/admin/class-migration.php
Comment (kienstra) [inline php/admin/class-migration.php:107]: This isn't translated, as it's the name of the plugin. Maybe it should be 😄

File: php/admin/class-migration.php
Comment (kienstra) [inline php/admin/class-migration.php:186]: If you don't want this to display in the Dashboard, it'd be easy to remove `'dashboard'` here

File: php/admin/class-migration.php
Comment (kienstra) [inline php/admin/class-migration.php]: This isn't translated, as it's the name of the plugin. Maybe it should be 😄

**Comment by kienstra** [comment]
#### Request for review

Hi @lukecarbis,
Could you please review this when you can? The [top comment](https://github.com/getblocklab/block-lab/pull/635#issue-441126183) is up to date.

Also, the notice doesn't have borders between the the buttons, like in the Migration UX document. 

Are those borders intended, or are they just from the document formatting?

borders-innotice

Thanks, Luke!

File: php/admin/class-migration.php
Comment (lukecarbis) [reply php/admin/class-migration.php]: I think it _should_ be translatable, but the translation should be "Genesis Custom Blocks" in every language.

So, yeah, switch to a `__()` here.

File: php/admin/class-migration.php
Comment (lukecarbis) [inline php/admin/class-migration.php]: @RobStino This page doesn't exist yet. Do we have anything we can chuck up there, or do we at least have this page marked as a to do on your go to market list? If so, let's make sure the URL is correct.

File: php/admin/class-migration.php
Comment (lukecarbis) [inline php/admin/class-migration.php]: Why use `wp_die()` over [`wp_send_json_error()`](https://developer.wordpress.org/reference/functions/wp_send_json_error/)?

**Review by lukecarbis — CHANGES_REQUESTED** [review]
Looking good Ryan! Couple of small questions, and minor change requested.

**Comment by lukecarbis** [comment]
Tested, and everything works as expected.

I'll let @RobStino comment on the design side of things, I suspect there's a few tweaks he'd like to make.

File: php/admin/class-migration.php
Comment (RobStino) [reply php/admin/class-migration.php]: No, this page doesn't exist yet, but I've got it marked as a to-do. I'll just need to make sure I use the correct URL.

File: php/admin/class-migration.php
Comment (kienstra) [reply php/admin/class-migration.php]: Good question, I'm fixing it 😄

Comment (kienstra) [reply php/admin/class-migration.php]: Fixed in https://github.com/getblocklab/block-lab/pull/635/commits/d2ce757313994d0d487d55cbc9c17afe8d9f4e95

**Comment by kienstra** [comment]
Hi @lukecarbis,
Thanks a lot for reviewing this, good points. They're applied above.

File: php/admin/class-migration.php
Comment (lukecarbis) [inline php/admin/class-migration.php]: This should be `wp_send_json_success()`, I think.

**Review by lukecarbis — APPROVED** [review]
👍 from me. Just one minor suggestion, but no need for further review from me after that.

Would love @RobStino's signoff on this before it's merged, though.

File: php/admin/class-migration.php
Comment (kienstra) [reply php/admin/class-migration.php]: Right, usually AJAX callbacks would use that.

This is a little different in that it doesn't send anything back to the [JS logic that called it](https://github.com/getblocklab/block-lab/pull/635/files#diff-682871fb6fb8d69d6b84890ae345f685R16).

Comment (lukecarbis) [reply php/admin/class-migration.php]: I understand that we wouldn't be using any success response. It might be more efficient to just `wp_die()`.

Either way, we should try to be consistent. Use `wp_die()` for both the error and success cases (as it was prior to my `wp_send_json_error()` suggestion), or use `wp_send_json_success()` here. We don't have to provide any data, we can call that function without any parameters.

The only difference is that the unused output would be:
`{'success':true}`

Instead of:
`1`

Comment (kienstra) [reply php/admin/class-migration.php:208]: Sounds good, how about just `wp_die()` for both?

Comment (kienstra) [reply php/admin/class-migration.php]: Good point that it's still sending `1` or `-1` in the response.

Comment (kienstra) [reply php/admin/class-migration.php]: https://github.com/getblocklab/block-lab/pull/635/commits/4e8167a4e178d46c4bd26852b73e6663183f1667 changes this to `wp_send_json_success()`

Comment (lukecarbis) [reply php/admin/class-migration.php]: Thanks for 4e8167a – looks good. Would have been happy with `wp_die()`, but I like this approach better, personally, because it feels consistent with other AJAX responses.

Comment (kienstra) [reply php/admin/class-migration.php]: Thanks, it is more like other AJAX calls 😄

**Comment by RobStino** [comment]
Hey Ryan. Thanks for the work on this.
If we can make some small adjustments to the design of the notice, that would be great. I'm keeping it super simple and inline with standard WP admin notifications.
Specific call outs:
- WP Admin Button styling to 2 primary actions
- Reduction of vertical spacing between text and "learn more" link
See below for my mock
image

The second stage of the notification for when someone clicks "not now" is good as is.

**Comment by kienstra** [comment]
Hi @RobStino,
Thanks, that looks good. I'll ping you here when it's applied.

**Comment by kienstra** [comment]
Hi @RobStino,
Thanks, I think your suggestion looks better.

How does it look with https://github.com/getblocklab/block-lab/pull/635/commits/0f1c6ea248101ac2b46e37ef2c995cb7af5b9573?

### Narrow screen

narrow-width

### Wider screen

wider-screen

There's no change in the 'Not now' notice:

not-now-notice

Also, I [added](https://github.com/getblocklab/block-lab/pull/635/files#diff-24db437d8589eed1f645b12ea7e577a0R92) a `@todo` as a note to myself to check the link.

**Comment by lukecarbis** [comment]
I think it needs a touch more padding to the left of the Not Now button.

**Comment by RobStino** [comment]
Hey. Could you add 16px more margin to the left of the "Not Now" button?

**Comment by kienstra** [comment]
Sure!

**Comment by kienstra** [comment]
@RobStino,
Good idea. How does it look now with https://github.com/getblocklab/block-lab/pull/635/commits/4137f1addd19f0bb1f2277fdd8b05e910bdecc2d?

notice-margin-left-full-screen

**Review by RobStino — APPROVED** [review]
Looks good to me! Thanks for the work.

**Comment by kienstra** [comment]
Nice, thanks for reviewing it!

**Pull request merged**
